pkg/templates/nstemplatetiers/nstemplatetier_generator.go

```diff
@@ -28,16 +32,33 @@ func CreateOrUpdateResources(ctx context.Context, s *runtime.Scheme, client runt
 	if err != nil {
 		return errors.Wrapf(err, "unable to load templates")
 	}
+
+	paths, err := templates.GetAllFileNames(&nstemplatetierFS, root)
+	if err != nil {
+		return err
+	}
+	if len(paths) == 0 {
+		return fmt.Errorf("could not find any user templates")
+	}
+
 	files := map[string][]byte{}
-	for _, name := range assets.Names() {
-		if name == "metadata.yaml" {
+	for _, name := range paths {
+		if name == "templates/nstemplatetiers/metadata.yaml" {
 			continue
 		}
-		content, err := assets.Asset(name)
+
+		parts := strings.Split(name, "/")
+		// skip any name that does not have 4 parts
```

Comment:
Again, this is assuming the structure of the fs (that we don't restrict in any sense) and therefore feels tightly coupled to the expected `deploy.NSTemplateTiersFS`.

Reply:
Okay so this is done here because this [function](https://github.com/codeready-toolchain/host-operator/pull/1165/files#diff-75fb59490e920da60f86dae8e59891d725272ef82d0515a32447036e0b576e40R65) in [toolchain common](https://github.com/codeready-toolchain/toolchain-common/blob/f85b77b1e378ee8ec2c367559e13b0d35470f007/pkg/template/nstemplatetiers/nstemplatetier_generator.go#L150) expects only the last two parts of the folder structure. 
and the test are written accordingly. 

moreover, it has to be in line with how our template folder structure is. similar is for user tiers, notification and others.

Reply:
I think that illustrates my point :) There's a function in `toolchain-common` that assumes the paths are 2 levels deep. Here, you assume that the paths are 4 levels deep but neither that function nor this function is enforcing that in any sense through the type-system or function signature.

This IMHO makes the code more fragile than it could be. If you removed the two parameters to this function, the assumption about 4 levels would suddenly be local to this function and the callers of this function would no longer need to care about it. This would IMHO improve the code.

Reply:
i updated the function PTAL